nemo/collections/nlp/modules/common/text_generation_utils.py

```diff
@@ -697,9 +697,11 @@ def sample_sequence_batch(
         all_generated_indices = None  # used to track all generated indices
         # Generate enough tokens for the longest sequence
         maxlen = tokens_to_generate + context_lengths.max().item()
-
         maxlen = inference_strategy.clip_max_len(maxlen)
 
+        maxlens = tokens_to_generate + context_lengths  # used to force EOS for shorter contexts in a batch.
```

Comment:
Do we need to also implement a similar logic in `tab_sample_sequence_batch()`? (by the way this function seems to already have drifted away a bit from `sample_sequence_batch()`, and I'm not sure this is intended)

Reply:
yes. let 's fix it in the `tab_sample_sequence_batch` which is used to tabular data generation.

Reply:
updated for tab_sample